Phoenix/PHSpace.m

```diff
@@ -222,9 +237,35 @@ - (void)removeWindows:(NSArray<PHWindow *> *)windows {
                                (__bridge CFArrayRef) @[@(self.identifier)]);
 }
 
+/**
+ * - https://github.com/kasper/phoenix/issues/348
+ * - https://github.com/koekeishiya/yabai/issues/2240#issuecomment-2116326165
+ * - https://github.com/ianyh/Amethyst/issues/1643#issuecomment-2132519682
+ */
+- (void)moveWindowsWithCompatId:(NSArray<PHWindow *> *)windows {
+    NSArray<NSNumber *> *ids = [self identifiersForWindows:windows];
+    NSUInteger windowCount = [ids count];
+    NSMutableData *windowIdSequence = [[NSMutableData alloc] initWithCapacity:windowCount * sizeof(CGWindowID)];
+    for (id item in ids) {
+        CGWindowID value = [item unsignedIntValue];
+        [windowIdSequence appendBytes:&value length:sizeof(CGWindowID)];
+    }
+
+    CGSConnectionID connection = CGSMainConnectionID();
+    CGSSpaceSetCompatID(connection, self.identifier, MoveWindowsCompatId);
+    CGSSetWindowListWorkspace(connection, (CGWindowID *)[windowIdSequence bytes], windowCount, MoveWindowsCompatId);
+    CGSSpaceSetCompatID(connection, self.identifier, 0x0);
+}
+
 - (void)moveWindows:(NSArray<PHWindow *> *)windows {
-    CGSMoveWindowsToManagedSpace(
-        CGSMainConnectionID(), (__bridge CFArrayRef)[self identifiersForWindows:windows], self.identifier);
+    if (![NSProcessInfo isOperatingSystemAtLeastSonoma145]) {
```

Comment:
I would still revert this conditional. Exception case first.

Reply:
IMO the exception case _is_ first, because versions of macOS < 14.5 will never increase, and versions of macOS >= 14.5 always increase 🤔 So to align with the intent of your comment, I'll keep the existing order.